AzureMonitor: Refresh documentation

joshhunt opened the pull request:
**What this PR does / why we need it**:

🎉  New Docs!!!

**Which issue(s) this PR fixes**:

<!--

- Automatically closes linked issue when the Pull Request is merged.

Usage: "Fixes #<issue number>", or "Fixes (paste link of issue)"

-->

Fixes #33713  
Fixes #33834

**Special notes for your reviewer**:

Still a work in progress, the old documentation is below the new one. Once it's all done I'll take it out of draft and remove the old stuff.

joshhunt committed:
AzureMonitor: Refresh documentation

joshhunt commented:
@achatterjee-grafana Would love your early thoughts on the structure so far. The copy is still a bit rough, and I need to add in the screenshots, but would love your feedback on the direction.

joshhunt committed:
logs/kusto

achatterjee-grafana commented on docs/sources/datasources/azuremonitor.md:
```diff
@@ -8,6 +8,189 @@ weight = 300
+
+## Azure Monitor settings
+
+To access Azure Monitor settings, hover your mouse over the **Configuration** (gear) icon, then click **Data Sources**, and then click the Azure Monitor data source. If you haven't already, you'll need to [add the Azure Monitor data source]({{< relref "add-a-data-source.md" >}}).
+
```
```suggestion
To access Azure Monitor settings, hover your mouse over the **Configuration** (gear) icon, click **Data Sources**, and then click the Azure Monitor data source. If you haven't already, you'll need to [add the Azure Monitor data source]({{< relref "add-a-data-source.md" >}}).

```

achatterjee-grafana commented on docs/sources/datasources/azuremonitor.md:
```diff
@@ -8,6 +8,189 @@ weight = 300
+
+To access Azure Monitor settings, hover your mouse over the **Configuration** (gear) icon, then click **Data Sources**, and then click the Azure Monitor data source. If you haven't already, you'll need to [add the Azure Monitor data source]({{< relref "add-a-data-source.md" >}}).
+
+You will need to create an app registration and service principal in Azure AD to authenticate the data source. See the [Azure documentation](https://docs.microsoft.com/en-us/azure/active-directory/develop/howto-create-service-principal-portal#get-tenant-and-app-id-values-for-signing-in) for details on this. Alternatively, if you are hosting Grafana in Azure (e.g. App Service, or Azure Virtual Mac ...
```
```suggestion
You must create an app registration and service principal in Azure AD to authenticate the data source. See the [Azure documentation](https://docs.microsoft.com/en-us/azure/active-directory/develop/howto-create-service-principal-portal#get-tenant-and-app-id-values-for-signing-in) for configuration details. Alternatively, if you are hosting Grafana in Azure (e.g. App Service, or Azure Virtual Machines) you can configure the Azure Monitor data source to use Managed Identity to securely authenticate without entering credentials into Grafana. Refer to [Configuring using Managed Identity](#todo) for more details.
```

achatterjee-grafana commented on docs/sources/datasources/azuremonitor.md:
```diff
@@ -8,6 +8,189 @@ weight = 300
+| Name                    | Description                                                                                                                                                                                                                                                                                                             |
+| ----------------------- | ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| Authentication          | Enables Managed Identity. See [Configuring using Managed Identity](#todo) for more details.                                                                                                                                                                                                                             |
+| Azure Cloud             | The national cloud for your Azure account. For most users this will be the default "Azure". For more details, see [](https://docs.microsoft.com/en-us/azure/active-directory/develop/authentication-national-cloud)                                                                                                     |
```
```suggestion
| Azure Cloud             | The national cloud for your Azure account. For most users, this is the default "Azure". For more information, see [](https://docs.microsoft.com/en-us/azure/active-directory/develop/authentication-national-cloud)                                                                                                     |
```

achatterjee-grafana reviewed (commented):
This is great restructuring! I added a few minor comments. Let us chat about how we can break down content into multiple topics. Otherwise we may end up with another long reference guide.

joshhunt committed:
finish up logs

joshhunt committed:
variables for log analytics

joshhunt committed:
Apply suggestions from code review

Co-authored-by: achatterjee-grafana <70489351+achatterjee-grafana@users.noreply.github.com>

joshhunt committed:
finish up main topics

joshhunt committed:
finish docs?

andresmgot commented on docs/sources/datasources/azuremonitor.md:
```diff
@@ -8,160 +8,116 @@ weight = 300
+1. For metrics that have multiple dimensions, you can split and filter further the returned metrics. For example, the Application Insights dependency calls metric supports returning multiple time series for successful vs unsuccessful calls.
 
-### Alias patterns for Metrics
+The options available will change depending on what is most relevent to the selected metric.
```
`relevent` -> `relevant`

andresmgot commented on docs/sources/datasources/azuremonitor.md:
```diff
@@ -8,160 +8,116 @@ weight = 300
+Azure Monitor Logs collects and organises log and performance data from [supported resources](https://docs.microsoft.com/en-us/azure/azure-monitor/monitor-reference) and makes many sources of data available to query together with the sophisticated [Kusto Query Language (KQL)](https://docs.microsoft.com/en-us/azure/data-explorer/kusto/query/).
 
-{{< figure src="/static/img/docs/v60/azuremonitor-alerting.png" class="docs-image--no-shadow" caption="Azure Monitor Alerting" >}}
+Wheras Azure Monitor Metrics can only store numeric data in a simplified structure, Logs can store different data types each with their own strucutre and can perform complexe analysis of data using KQL.
```
Wheras?

andresmgot reviewed (approved):
Great docs!

joshhunt commented on docs/sources/datasources/azuremonitor.md:
```diff
@@ -8,160 +8,116 @@ weight = 300
+### Querying Azure Monitor Metrics
 
-### Format legend keys with aliases for Metrics
+Azure Monitor Metrics collects numeric data from [supported resources](https://docs.microsoft.com/en-us/azure/azure-monitor/monitor-reference) and allows you to query them to investigate the health and utilisation of your resources to maximise availability and performance.
```
No, i think that's correct.

joshhunt commented on docs/sources/datasources/azuremonitor.md:
```diff
@@ -8,160 +8,116 @@ weight = 300
 
-The default legend formatting for the Metrics API is:
+Metrics are more lightweight than data in Azure Monitior Logs and only stores numeric data in a particular structure. Metrics is capable for supporting near real-time scenarios making it useful for fast detection of issues. Azure Monitor Logs can store a variety of different data types each with their own structure.
 
-`metricName{dimensionName=dimensionValue,dimensionTwoName=DimensionTwoValue}`
+> metrics screenshot
```
Logs is mentioned to contrast it to Metrics, but you're right - I'll rework it.

joshhunt committed:
typos and other review comments

joshhunt committed:
add link to sample arg queries

joshhunt committed:
split up azure docs

joshhunt committed:
workaround weird code duplication issue

joshhunt commented:
@grafana/docs-squad After a call with Jita I've split up the docs into a few more pages, and added in some screenshots. Running the docs locally was a really great way to visualise everything!

Would love some feedback on where this is now so I can merge it in :) :) 👍

sarahzinger reviewed (approved):
Looks good to me, I just wanted to add another little fix for https://github.com/grafana/grafana/issues/33834 if that's cool!

oddlittlebird commented on docs/sources/datasources/azuremonitor/_index.md:
```diff
@@ -0,0 +1,255 @@
+
+# Azure Monitor data source
+
+Grafana includes built-in support for Azure Monitor, the Azure service to maximise availability and performance of your applications and services in the Azure Cloud. The Azure Monitor data source supports visualising data from three Azure services:
```
Yes, we do! Good catch!

joshhunt committed:
Update docs/sources/datasources/azuremonitor/template-variables.md

Co-authored-by: Sarah Zinger <sarahzinger@users.noreply.github.com>

joshhunt commented on docs/sources/datasources/azuremonitor/template-variables.md:
```diff
@@ -0,0 +1,57 @@
+| `MetricNamespace(aResourceGroup, aNamespace, aResourceName)`                 | Returns a list of metric namespaces.                                                                   |
+| `MetricNamespace(subscriptionID, aResourceGroup, aNamespace, aResourceName)` | Returns a list of metric namespaces for a specified subscription.                                      |
+| `MetricNames(aResourceGroup, aNamespace, aResourceName)`                     | Returns a list of metric names.                                                                        |
+| `MetricNames(subscriptionID, aResourceGroup, aNamespace, aResourceName)`     | Returns a list of metric names for a specified subscription.                                           |
```
good shout - thanks!!

achatterjee-grafana commented on docs/sources/datasources/azuremonitor/_index.md:
```diff
@@ -0,0 +1,255 @@
+
+# Azure Monitor data source
+
+Grafana includes built-in support for Azure Monitor, the Azure service to maximise availability and performance of your applications and services in the Azure Cloud. The Azure Monitor data source supports visualising data from three Azure services:
```
```suggestion
Grafana includes built-in support for Azure Monitor, the Azure service to maximize the availability and performance of your applications and services in the Azure Cloud. The Azure Monitor data source supports visualizing data from three Azure services:
```

achatterjee-grafana commented on docs/sources/datasources/azuremonitor/_index.md:
```diff
@@ -0,0 +1,255 @@
+
+- **Azure Monitor Metrics** to collect numeric data from resources in your Azure account
+- **Azure Monitor Logs** to collect log and performance data from your Azure account, and query using the powerful Kusto Language
+- **Azure Resource Graph** to quickly query your Azure resources across subscriptions
```
Add a period "." at the end of each bullet.

achatterjee-grafana commented on docs/sources/datasources/azuremonitor/_index.md:
```diff
@@ -0,0 +1,255 @@
+
+## Azure Monitor configuration
+
+To access Azure Monitor configuration, hover your mouse over the **Configuration** (gear) icon, click **Data Sources**, and then click the Azure Monitor data source. If you haven't already, you'll need to [add the Azure Monitor data source]({{< relref "add-a-data-source.md" >}}).
```
```suggestion
To access Azure Monitor configuration, hover your mouse over the **Configuration** (gear) icon, click **Data Sources**, and then select the Azure Monitor data source. If you haven't already, you'll need to [add the Azure Monitor data source]({{< relref "add-a-data-source.md" >}}).
```

achatterjee-grafana commented on docs/sources/datasources/azuremonitor/_index.md:
```diff
@@ -0,0 +1,255 @@
+
+- **Metrics** for [Azure Monitor Metrics](https://docs.microsoft.com/en-us/azure/azure-monitor/essentials/data-platform-metrics)
+- **Logs** for [Azure Monitor Logs](https://docs.microsoft.com/en-us/azure/azure-monitor/logs/data-platform-logs)
+- [**Azure Resource Graph**](https://docs.microsoft.com/en-gb/azure/governance/resource-graph/overview)
```
Azure Resource Graph should be bold.

achatterjee-grafana replied:
Also, I suggest adding hyperlinks to topics like "Querying Azure Monitor Metrics", "Querying Azure Monitor Logs" and "Querying with Azure Resource Graph" upfront. Otherwise, users may not know the Querying Azure Monitor Logs section exists.

joshhunt replied:
Azure Resource Graph is bold 😅

Good suggestion for the links though

achatterjee-grafana reviewed (commented):
Added some copy-edit suggestions.

joshhunt committed:
Apply suggestions from code review

Co-authored-by: achatterjee-grafana <70489351+achatterjee-grafana@users.noreply.github.com>

joshhunt committed:
feedback
